principles/solid/dependencyInjection.md

```diff
@@ -1,22 +1,88 @@
 ## Dependency Injection
 
-[TODO]
+It states that dependencies must be on the abstractions, not the concretions. That is to say:
+
+- High level modules should not depend on low level modules. Both should depend on abstractions.  
+- Abstractions should not depend on details. Details should depend on abstractions.
+
+At some point our program or application will be made up of many modules. When this happens, it is when we must use dependency injection, which will allow us to control the functionalities from a specific place instead of having them scattered throughout the program. In addition, this isolation will allow us to test much more easily.
 
 ### Bad example
 
-[TODO]
+Suppose we have a class to access data, and we do it through a DB:
+
+```JAVA:
+    class DatabaseService{
+        //...
+        void getData(){ //... }
+    }
+
+    class DataAccess {
+
+        private DatabaseService databaseService;
+
+        public DataAccess(DatabaseService databaseService){
```

Comment:
Here is using dependency Injection. Change the example

Reply:
Profe pero es el principio de Inyeccion de dependencias?